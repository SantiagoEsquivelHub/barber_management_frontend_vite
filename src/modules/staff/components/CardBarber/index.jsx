import React from 'react'
import PropTypes from 'prop-types';
import iconBarber from '../../../../assets/images/icono.png'

/*Componente usado para mostrar cada uno de los barberos, muestra el nombre, la imagen, el correo, el telÃ©fono y su estado*/

const CardBarber = ({ nombre, correo, telefono, estado, url, id }) => {
    return (
        <div>
            <div id={id} className="cardBarber col-12">

                <li className="ant-list-item"><div class="ant-list-item-meta">
                    <div className="ant-list-item-meta-avatar">
                        <span className="ant-avatar ant-avatar-circle ant-avatar-image">
                            <img src={url} />
                        </span>
                    </div>
                    <div className="ant-list-item-meta-content">
                        <h4 className="ant-list-item-meta-title" >
                            <a href={`/staff/${id}`} id={id}>{nombre}</a>
                        </h4>
                    </div>
                    <div className="ant-list-item-meta-content">
                        <div className="ant-list-item-meta-description correo">{correo}</div>
                    </div>
                    <div className="ant-list-item-meta-content">
                        <div className="ant-list-item-meta-description telefono">{telefono}</div>
                    </div>
                </div>
                    <div className={estado == 1 ? 'activo' : 'deshabilitado'}>{estado == 1 ? 'Activo' : 'Deshabilitado'}</div>
                    <ul className="ant-list-item-action">
                        <li>
                            <img src={iconBarber} className="icono_barber" />
                        </li>
                    </ul>
                </li>
            </div>
        </div>
    )
}

CardBarber.propTypes = {
    nombre: PropTypes.string.isRequired,
    correo: PropTypes.string.isRequired,
    telefono: PropTypes.string.isRequired,
    estado: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
}
export default CardBarber;
