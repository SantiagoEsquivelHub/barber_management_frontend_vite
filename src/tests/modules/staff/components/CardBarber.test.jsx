import { render } from "@testing-library/react"
import CardBarber from "../../../../modules/staff/components/CardBarber"
import icon from '../../../../assets/images/UserPhoto.png';

describe('Pruebas en <CardBarber />', () => {


    const nombre = 'Santiago';
    const correo = 'santi@correo.com';
    const telefono = 123456789;
    const estado = 1;
    const id = 10;

    test('Debe hacer match con el Snapshot', () => {
        const { container } = render(<CardBarber nombre={nombre} correo={correo} telefono={telefono} estado={estado} id={id} url={icon} />)
        expect(container).toMatchSnapshot();
    })

})