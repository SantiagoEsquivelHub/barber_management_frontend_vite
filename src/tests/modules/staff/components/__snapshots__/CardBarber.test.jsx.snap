// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CardBarber /> Debe hacer match con el Snapshot 1`] = `
<div>
  <div>
    <div
      class="cardBarber col-12"
      id="10"
    >
      <li
        class="ant-list-item"
      >
        <div
          class="ant-list-item-meta"
        >
          <div
            class="ant-list-item-meta-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
            >
              <img
                src=""
              />
            </span>
          </div>
          <div
            class="ant-list-item-meta-content"
          >
            <h4
              class="ant-list-item-meta-title"
            >
              <a
                href="/staff/10"
                id="10"
              >
                Santiago
              </a>
            </h4>
          </div>
          <div
            class="ant-list-item-meta-content"
          >
            <div
              class="ant-list-item-meta-description correo"
            >
              santi@correo.com
            </div>
          </div>
          <div
            class="ant-list-item-meta-content"
          >
            <div
              class="ant-list-item-meta-description telefono"
            >
              123456789
            </div>
          </div>
        </div>
        <div
          class="activo"
        >
          Activo
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <img
              class="icono_barber"
              src=""
            />
          </li>
        </ul>
      </li>
    </div>
  </div>
</div>
`;
